src/hostapi/pulseaudio/pa_hostapi_pulseaudio.c

```diff
@@ -0,0 +1,1249 @@
+    if( !hostapi->pulseaudioDeviceNames[hostapi->deviceCount] &&
+        !hostapi->deviceInfoArray[hostapi->deviceCount].name)
+    {
+       PA_PULSEAUDIO_SET_LAST_HOST_ERROR(0,
+                                         "Can't alloc memory"); 
+       return paInsufficientMemory;
+    }
+    /*PA_UNLESS(hostapi->pulseaudioDeviceNames[hostapi->deviceCount] =
+              (char *) PaUtil_GroupAllocateMemory(hostapi->allocations,
+                                                  l_iRealNameLen),
+              paInsufficientMemory);*/
+
+    /* PA_UNLESS(hostapi->deviceInfoArray[hostapi->deviceCount].name =
+              (char *) PaUtil_GroupAllocateMemory(hostapi->allocations,
+                                                  l_iDeviceNameLen),
+              paInsufficientMemory); */
+
+    strncpy((char *)
+            hostapi->pulseaudioDeviceNames[hostapi->deviceCount],
+            realName, l_iRealNameLen);
+
+    strncpy((char *) hostapi->deviceInfoArray[hostapi->deviceCount].name,
+          interfaceName, l_iDeviceNameLen);
+
+    l_ptrName = (char *)hostapi->deviceInfoArray[hostapi->deviceCount].name + l_iDeviceNameLen;
+    l_ptrName = '\0';
+    hostapi->pulseaudioDeviceNames[hostapi->deviceCount][l_iRealNameLen] = '\0';
+
+    hostapi->deviceInfoArray[hostapi->deviceCount].maxInputChannels = inputChannels;
+    hostapi->deviceInfoArray[hostapi->deviceCount].maxOutputChannels = outputChannels;
+    hostapi->deviceInfoArray[hostapi->deviceCount].defaultLowInputLatency = defaultLowInputLatency;
+    hostapi->deviceInfoArray[hostapi->deviceCount].defaultLowOutputLatency = defaultLowOutputLatency;
+    hostapi->deviceInfoArray[hostapi->deviceCount].defaultHighInputLatency = defaultHighOutputLatency;
+    hostapi->deviceInfoArray[hostapi->deviceCount].defaultHighOutputLatency = defaultHighOutputLatency;
+    hostapi->deviceInfoArray[hostapi->deviceCount].defaultSampleRate = defaultSampleRate;
+    hostapi->deviceCount++;
+}
+
+void PulseAudioSinkListCb(
+    pa_context * c,
+    const pa_sink_info * l,
+    int eol,
+    void *userdata
+)
+{
+    PaPulseAudioHostApiRepresentation *l_ptrHostApi =
+        (PaPulseAudioHostApiRepresentation *) userdata;
+    PaError result = paNoError;
+    char *l_strName = NULL;
+
+    assert(c);
+
+    /* If this is null we have big problems and we probably are out of memory */
+    if (!c)
+    {
+        PA_PULSEAUDIO_SET_LAST_HOST_ERROR(0,
+                                          "PulseAudioSinkListCb: Out of memory");
+        goto error;
+    }
+
+    /* If eol is set to a positive number, you're at the end of the list */
+    if (eol > 0)
+    {
+        goto error;
+    }
+
+    l_ptrHostApi->deviceInfoArray[l_ptrHostApi->deviceCount].structVersion = 2;
+    l_ptrHostApi->deviceInfoArray[l_ptrHostApi->deviceCount].hostApi =
+        l_ptrHostApi->hostApiIndex;
+
+    l_strName = (char *) l->name;
+
+    if (l->description != NULL)
+    {
+        l_strName = (char *) l->description;
+    }
```

Comment:
User here the parameter names and types of  _PulseAudioAddAudioDevice()
const char *interfaceName
const char *realName

By the way are these names reasonable, they are not uses in the Pa and not in the Pulse API.